### PR TITLE
Added PWM and ADC skeleton classes etc.

### PR DESCRIPTION
**Pull request by networkfusion**
## Description
Added all of the options and classes required to implement PWM and ADC.

TODO (with help):
- Add required native methods.
- Adjust mcuconf.h accordingly.

## Motivation and Context
Needed for required functionality. Works towards finishing the nanoframework-class-libraries ADC and PWM functionality contained in the respective pull request

## How Has This Been Tested? (if applicable)
Builds locally.
Not yet added to Travis (as default options set to off.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by josesimoes** [comment]
@networkfusion please do a git rebase on this branch so that only your changes show on the commit history.

**Comment by networkfusion** [comment]
I thought I did that:

git rebase master

What am I doing wrong?

**Comment by josesimoes** [comment]
I'm no Git expert....  try upstream/master...

File: targets/CMSIS-OS/ChibiOS/nanoCLR/Windows.Devices.Adc/win_dev_adc_native.cpp [hunk 1]
```diff
@@ -0,0 +1,2 @@
+#include "win_dev_adc_native.h"
```
Comment (josesimoes) [inline targets/CMSIS-OS/ChibiOS/nanoCLR/Windows.Devices.Adc/win_dev_adc_native.cpp]: Please add the copyright header.

File: targets/CMSIS-OS/ChibiOS/nanoCLR/Windows.Devices.Adc/win_dev_adc_native.h [hunk 2]
```diff
@@ -0,0 +1,20 @@
+//-----------------------------------------------------------------------------
```
Comment (josesimoes) [inline targets/CMSIS-OS/ChibiOS/nanoCLR/Windows.Devices.Adc/win_dev_adc_native.h]: Please replace with nF copyright header.

File: targets/CMSIS-OS/ChibiOS/nanoCLR/Windows.Devices.Adc/win_dev_adc_native.h [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+//
```
Comment (josesimoes) [inline targets/CMSIS-OS/ChibiOS/nanoCLR/Windows.Devices.Adc/win_dev_adc_native.h]: Please replace with nF copyright header.

File: targets/CMSIS-OS/ChibiOS/nanoCLR/Windows.Devices.Adc/win_dev_adc_native_Windows_Devices_ADC_AdcDevice.cpp [hunk 4]
```diff
@@ -0,0 +1,54 @@
+//-----------------------------------------------------------------------------
```
Comment (josesimoes) [inline targets/CMSIS-OS/ChibiOS/nanoCLR/Windows.Devices.Adc/win_dev_adc_native_Windows_Devices_ADC_AdcDevice.cpp]: Please replace with nF copyright header.

File: targets/CMSIS-OS/ChibiOS/nanoCLR/Windows.Devices.Pwm/win_dev_pwm_native.cpp [hunk 5]
```diff
@@ -0,0 +1,2 @@
+#include "win_dev_pwm_native.h"
```
Comment (josesimoes) [inline targets/CMSIS-OS/ChibiOS/nanoCLR/Windows.Devices.Pwm/win_dev_pwm_native.cpp]: Please add nF copyright header.

File: targets/CMSIS-OS/ChibiOS/nanoCLR/Windows.Devices.Pwm/win_dev_pwm_native.h [hunk 6]
```diff
@@ -0,0 +1,20 @@
+//-----------------------------------------------------------------------------
```
Comment (josesimoes) [inline targets/CMSIS-OS/ChibiOS/nanoCLR/Windows.Devices.Pwm/win_dev_pwm_native.h]: Please replace with nF copyright header.

File: targets/CMSIS-OS/ChibiOS/nanoCLR/Windows.Devices.Pwm/win_dev_pwm_native_Windows_Devices_PWM_PwmDevice.cpp [hunk 7]
```diff
@@ -0,0 +1,43 @@
+//-----------------------------------------------------------------------------
```
Comment (josesimoes) [inline targets/CMSIS-OS/ChibiOS/nanoCLR/Windows.Devices.Pwm/win_dev_pwm_native_Windows_Devices_PWM_PwmDevice.cpp]: Please replace with nF copyright header.

**Comment by networkfusion** [comment]
@josesimoes do you want me to update the headers in the similar i2c classes as well... It is obviously where I got these ones from?...

**Comment by josesimoes** [comment]
> do you want me to update the headers in the similar i2c classes as well 

That has to be done for sure, but I fear that right now that could cause merge issues in the files that @MikroBusNet is working. Let's leave that on pause for now. (thanks for asking 👍 )

**Comment by networkfusion** [comment]
oops, I pushed that by mistake (touching the I2C files...) I will revert and squash

**Pull request merged**